src/test/ui/dyn-star/const.rs

```diff
@@ -6,7 +6,7 @@ use std::fmt::Debug;
 
 fn make_dyn_star() {
     let i = 42usize;
-    let dyn_i: dyn* Debug = i as dyn* Debug;
+    let dyn_i: dyn* Debug = i;
```

Comment:
It's nice to be able to get rid of the stutter here!

It might be worth keeping around a test that does `let dyn_i = i as dyn* Debug` just to make sure we have both forms, but it's probably doesn't actually add anything useful in the way of test coverage.

Reply:
Can do!